apps/rush-lib/src/logic/Git.ts

```diff
@@ -3,76 +3,100 @@
 
 import gitInfo = require('git-repo-info');
 import * as child_process from 'child_process';
+import * as os from 'os';
 
-import { PublishUtilities } from './PublishUtilities';
+import { Utilities } from '../utilities/Utilities';
+import { AlreadyReportedError } from '../utilities/AlreadyReportedError';
+import { GitEmailPolicy } from './policy/GitEmailPolicy';
+import { RushConfiguration } from '../api/RushConfiguration';
 
-export class Git {
-  private _targetBranch: string | undefined;
-
-  public static detectIfGitIsSupported(): boolean {
-    const command: string = process.platform === 'win32' ? 'where' : 'which';
-    const result: child_process.SpawnSyncReturns<string> = child_process.spawnSync(command, ['git']);
-
-    if (result.status !== 0) {
-      return false;
-    }
+interface IResultOrError<TResult> {
+  error?: Error;
+  result?: TResult;
+}
 
-    try {
-      return !!gitInfo().sha;
-    } catch (e) {
-      return false; // Unexpected, but possible if the .git directory is corrupted.
+export class Git {
+  private static _hasGit: boolean | undefined = undefined;
+  private static _gitPath: string | undefined;
+
+  public static getGitPath(): string | undefined {
```

Comment:
>getGitPath [](start = 16, length = 10)

Document that this returns undefined if the Git tool is not installed (or discoverable from the PATH)